### PR TITLE
Use a custom dazel name to avoid name collisions

### PR DESCRIPTION
**Pull request by CodingCanuck**
Progress towards https://github.com/reboot-dev/respect-integration/issues/142

### CODE REVIEW

**Review by rjhuijsman — APPROVED** [review]

File: .dazelrc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Source: https://github.com/nadirizr/dazel/blob/master/dazel.py#L600-L601
+
+import os.path
+
+# Instead of using the host's Bazel cache (at ~/.cache/bazel), use a directory
+# that Dazel has to itself. This is essential to avoid a recurring problem with
+# compilation failures due to corrupt caches. Even Dazel's maintainers think
+# using the same cache is a mistake:
+#   https://github.com/nadirizr/dazel/issues/43
+DAZEL_BAZEL_USER_OUTPUT_ROOT = os.path.expanduser("~/.cache/dazel/tests")
```
Comment (benh) [inline .dazelrc:12]: Is `tests` here necessary? If so, can we document why? Otherwise can we just do `~/.cache/dazel/tests`?

File: .dazelrc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+DAZEL_DOCKERFILE = "Dockerfile.dazel"
+DAZEL_VOLUMES = ["/var/run/docker.sock:/var/run/docker.sock"]
```
Comment (benh) [inline .dazelrc:15]: Can we leave a comment as to why we need/want to expose the host's Docker? IIUC the Ubuntu build which uses `dazel` has been working before this so it's not clear why we need to add it now and a would be great!

**Review by benh — COMMENTED** [review]
Was catching up on merged PRs and had a few questions on this one!

File: .dazelrc (hunk 2 above)
Comment (CodingCanuck) [reply .dazelrc:15]: You removed this in https://github.com/3rdparty/eventuals/pull/314 ! :D

File: .dazelrc (hunk 1 above)
Comment (CodingCanuck) [reply .dazelrc:12]: Nope, and you cleaned this up in https://github.com/3rdparty/eventuals/pull/315 ! :D

**Review by CodingCanuck — COMMENTED** [review]
Closing out old comments for posterity

**Pull request merged**